test/optimisation/Optimisation.jl

```diff
@@ -0,0 +1,557 @@
+@testset "optimisation/Optimisation.jl" begin
+
+    # Issue: https://discourse.julialang.org/t/two-equivalent-conditioning-syntaxes-giving-different-likelihood-values/100320
+    @testset "OptimizationContext" begin
+        # Used for testing how well it works with nested contexts.
+        struct OverrideContext{C,T1,T2} <: DynamicPPL.AbstractContext
```

Comment:
I wonder if this could maybe be replaced with `DynamicPPL.check_model_and_trace` and look at the captured trace :thinking: 

Don't do this for this PR, but might be interesting to consider for future testing.